lncrawl/core/downloader.py

```diff
@@ -109,7 +109,9 @@ def _fetch_content_image(app, url, image_file):
         try:
             img = app.crawler.download_image(url)
             os.makedirs(os.path.dirname(image_file), exist_ok=True)
-            img.convert("RGB").save(image_file, "JPEG")
+            if img.mode in ("RGBA", "P"):
```

Comment:
isn't it more safe to check not equal?

Reply:
Yes I don't know why I did that. I tested theses modes supported by pillow :
```py
img = Image.open('test/test.jpg')
img.convert("L").save('test/test/jpeg-to-l.jpg', 'JPEG', optimize=True)
# img.convert("1").save('test/test/jpeg-to-1.jpg', 'JPEG', optimize=True) # file too big
# img.convert("P").save('test/test/jpeg-to-p.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode P as JPEG
img.convert("RGB").save('test/test/jpeg-to-rgb.jpg', 'JPEG', optimize=True)
# img.convert("RGBA").save('test/test/jpeg-to-rgba.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode RGBA as JPEG
# img.convert("CMYK").save('test/test/jpeg-to-cmyk.jpg', 'JPEG', optimize=True) # file too big
img.convert("YCbCr").save('test/test/jpeg-to-ycbcr.jpg', 'JPEG', optimize=True)
# img.convert("I").save('test/test/jpeg-to-i.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode I as JPEG
# img.convert("F").save('test/test/jpeg-to-f.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode F as JPEG
# img.convert("LA").save('test/test/jpeg-to-la.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode LA as JPEG
img.convert("RGBX").save('test/test/jpeg-to-rgbx.jpg', 'JPEG', optimize=True)
# img.convert("RGBa").save('test/test/jpeg-to-rgba.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode RGBa as JPEG
# img.convert("HSV").save('test/test/jpeg-to-hsv.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode HSV as JPEG
# img.convert("LAB").save('test/test/jpeg-to-lab.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode LAB as JPEG
# img.convert("LABa").save('test/test/jpeg-to-laba.jpg', 'JPEG', optimize=True) # ValueError: cannot write mode LABa as JPEG
```

Reply:
Your test indicates that not all formats can be saved as JPEG. But we are converting other formats to RGB first before saving to JPEG. Can you test if all formats can be converted to RGB?

Reply:
Good idea, it did not check for it before and I through it would work fine, but it does not for RGBa :
```py
Image.new('L', (100, 100)).convert('RGB')
Image.new('1', (100, 100)).convert('RGB')
Image.new('P', (100, 100)).convert('RGB')
Image.new('RGB', (100, 100)).convert('RGB')
Image.new('RGBA', (100, 100)).convert('RGB')
Image.new('CMYK', (100, 100)).convert('RGB')
Image.new('YCbCr', (100, 100)).convert('RGB')
Image.new('I', (100, 100)).convert('RGB')
Image.new('F', (100, 100)).convert('RGB')
Image.new('LA', (100, 100)).convert('RGB')
Image.new('RGBX', (100, 100)).convert('RGB')
Image.new('HSV', (100, 100)).convert('RGB')
Image.new('LAB', (100, 100)).convert('RGB')

# Image.new('RGBa', (100, 100)).convert('RGB') # ValueError: conversion from RGBa to RGB not supported
Image.new('RGBa', (100, 100)).convert('RGBA').convert('RGB')
```